### PR TITLE
feat: add an ability to import the Git Configuration

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
This PR adds an ability to import the Git Configuration.

### What issues does this PR fix or reference?
fixes https://github.com/eclipse-che/che/issues/22782

### Is it tested? How?

The [demo video](https://youtu.be/ka4cIEcE1_U).

Знімок екрана 2024-06-24 о 16 27 08

Знімок екрана 2024-06-24 о 16 21 52

### CODE REVIEW

**Review by ibuziuk — APPROVED** [review]
LGTM, but let's merge it for 7.89.x after 7.88.0 release is out 
we would also need review from @akurinnoy and @dmytro-ndp / @artaleks9

**Comment by artaleks9** [comment]
Verified on Che with `quay.io/eclipse/che-dashboard:pr-1136` - **works properly**

**Review by artaleks9 — APPROVED** [review]

File: packages/dashboard-frontend/package.json [hunk 1]
```diff
@@ -48,6 +48,7 @@
     "inversify-inject-decorators": "^3.1.0",
     "inversify-react": "^1.1.0",
     "js-yaml": "^4.1.0",
+    "multi-ini": "^2.3.2",
```
Comment (akurinnoy) [inline packages/dashboard-frontend/package.json:51]: Please regenerate lists of used dependencies

**Comment by artaleks9** [comment]
It seems, the `dashboard-happy-path` test is fail by an infrastructure problem related to Openshift-CI:
```
Could not retrieve mirrorlist http://mirrorlist.centos.org/?release=7&arch=x86_64&repo=os&infra=container error was
14: curl#6 - "Could not resolve host: mirrorlist.centos.org; Unknown error"
```

File: packages/dashboard-frontend/package.json (hunk 1 above)
Comment (olexii4) [reply packages/dashboard-frontend/package.json:51]: I did it. We reuse the same library as we have for the server-side.

File: packages/dashboard-frontend/src/pages/UserPreferences/GitConfig/AddModal/__tests__/index.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  });
+
+  it('should handle click on Close button', () => {
+    renderComponent(true);
+
+    const closeButton = screen.queryByRole('button', { name: 'Close' });
+    expect(closeButton).toBeTruthy();
+
+    userEvent.click(closeButton!);
+    expect(mockOnClose).toBeCalledTimes(1);
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/pages/UserPreferences/GitConfig/AddModal/__tests__/index.spec.tsx:76]: I have fixed it.

File: packages/dashboard-frontend/src/pages/UserPreferences/GitConfig/AddModal/GitConfigForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  }
+
+  public shouldComponentUpdate(nextProps: Readonly<Props>, nextState: Readonly<State>): boolean {
+    const { gitConfig, validated } = this.state;
+    const { gitConfig: nextGitConfig, validated: nextValidated } = nextState;
+
+    return gitConfig !== nextGitConfig || validated !== nextValidated;
+  }
+
+  private getGitConfig(gitConfigStr: string): GitConfigStore.GitConfig {
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/pages/UserPreferences/GitConfig/AddModal/GitConfigForm/index.tsx]: I have fixed it.

File: packages/dashboard-frontend/src/pages/UserPreferences/GitConfig/AddModal/GitConfigForm/index.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+          validated={validated}
+        >
+          <GitConfigImport
+            fieldId="git-configuration"
+            content={content}
+            fileNamePlaceholder="Upload the .gitconfig file"
+            textAreaPlaceholder="Or paste the Git Configuration here"
+            validated={validated}
+            onChange={(gitConfig, isUpload) => this.onChange(gitConfig, isUpload)}
+          />
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/pages/UserPreferences/GitConfig/AddModal/GitConfigForm/index.tsx]: I have fixed it.

File: packages/dashboard-frontend/src/pages/UserPreferences/GitConfig/AddModal/GitConfigForm/index.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  }
+
+  private getGitConfig(gitConfigStr: string): GitConfigStore.GitConfig {
+    const parser = new ini.Parser();
+    const gitConfigLines = gitConfigStr.split(/\r?\n/);
+    const gitConfig = parser.parse(gitConfigLines);
+    return gitConfig as GitConfigStore.GitConfig;
+  }
+
+  private getContent(gitConfig: GitConfigStore.GitConfig | undefined): string | undefined {
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/pages/UserPreferences/GitConfig/AddModal/GitConfigForm/index.tsx]: I have fixed it.

File: packages/dashboard-frontend/src/pages/UserPreferences/GitConfig/AddModal/GitConfigForm/__tests__/index.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+          data-testid="submit-valid-git-config"
+          type="button"
+          onClick={() =>
+            props.onChange('[user]\n\tname = User One\n\temail = user-1@chetest.com\n', true)
+          }
+        />
+      </div>
+    );
+  },
+}));
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/pages/UserPreferences/GitConfig/AddModal/GitConfigForm/__tests__/index.spec.tsx]: I have fixed it.

File: packages/dashboard-frontend/src/pages/UserPreferences/GitConfig/AddModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+          onClick={() => this.handleSaveGitConfig()}
+        >
+          Add
+        </Button>
+        <Button variant={ButtonVariant.link} onClick={() => this.handleCloseModal()}>
+          Cancel
+        </Button>
+      </React.Fragment>
+    );
+  }
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/pages/UserPreferences/GitConfig/AddModal/index.tsx]: I have fixed it.

**Review by akurinnoy — APPROVED** [review]

**Pull request merged**